handler/rfc8693/access_token_type_handler.go

```diff
@@ -0,0 +1,210 @@
+	} else if err := c.CoreStrategy.ValidateAccessToken(ctx, or, token); err != nil {
+		return nil, nil, err
+	}
+
+	subjectTokenClientID := or.GetClient().GetID()
+	// forbid original subjects client to exchange its own token
+	if client.GetID() == subjectTokenClientID {
+		return nil, nil, errors.WithStack(fosite.ErrRequestForbidden.WithHint("Clients are not allowed to perform a token exchange on their own tokens."))
+	}
+
+	// Check if the client is allowed to exchange this token
+	if subjectTokenClient, ok := or.GetClient().(Client); ok {
+		allowed := subjectTokenClient.TokenExchangeAllowed(client)
+		if !allowed {
+			return nil, nil, errors.WithStack(fosite.ErrRequestForbidden.WithHintf(
+				"The OAuth 2.0 client is not permitted to exchange a subject token issued to client %s", subjectTokenClientID))
+		}
+	}
+
+	// Scope check
+	for _, scope := range request.GetRequestedScopes() {
+		if !c.ScopeStrategy(or.GetGrantedScopes(), scope) {
+			return nil, nil, errors.WithStack(fosite.ErrInvalidScope.WithHintf("The subject token is not granted \"%s\" and so this scope cannot be requested.", scope))
+		}
+	}
+
+	// Convert to flat session with only access token claims
+	tokenObject := session.AccessTokenClaimsMap()
+	tokenObject["client_id"] = or.GetClient().GetID()
+	tokenObject["scope"] = or.GetGrantedScopes()
+	tokenObject["aud"] = or.GetGrantedAudience()
+
+	return or.GetSession(), tokenObject, nil
+}
+
+func (c *AccessTokenTypeHandler) issue(ctx context.Context, request fosite.AccessRequester, response fosite.AccessResponder) error {
+	request.GetSession().SetExpiresAt(fosite.AccessToken, time.Now().UTC().Add(c.AccessTokenLifespan))
+
+	token, signature, err := c.CoreStrategy.GenerateAccessToken(ctx, request)
+	if err != nil {
+		return err
+	} else if err := c.Storage.CreateAccessTokenSession(ctx, signature, request.Sanitize([]string{})); err != nil {
+		return err
+	}
+
+	issueRefreshToken := c.canIssueRefreshToken(request)
+	if issueRefreshToken {
+		request.GetSession().SetExpiresAt(fosite.RefreshToken, time.Now().UTC().Add(c.RefreshTokenLifespan).Round(time.Second))
+		refresh, refreshSignature, err := c.CoreStrategy.GenerateRefreshToken(ctx, request)
+		if err != nil {
+			return errors.WithStack(fosite.ErrServerError.WithDebug(err.Error()))
+		}
+
+		if refreshSignature != "" {
+			if err := c.Storage.CreateRefreshTokenSession(ctx, refreshSignature, request.Sanitize([]string{})); err != nil {
+				if rollBackTxnErr := storage.MaybeRollbackTx(ctx, c.Storage); rollBackTxnErr != nil {
+					err = rollBackTxnErr
+				}
+				return errors.WithStack(fosite.ErrServerError.WithDebug(err.Error()))
+			}
+		}
+
+		response.SetExtra("refresh_token", refresh)
+	}
+
+	response.SetAccessToken(token)
+	response.SetTokenType("bearer")
+	response.SetExpiresIn(c.getExpiresIn(request, fosite.AccessToken, c.AccessTokenLifespan, time.Now().UTC()))
+	response.SetScopes(request.GetGrantedScopes())
+
```

Comment:
shall we add `issued_token_type` just like here https://github.ibm.com/strata/ristretto-core/blob/master/fositex/handler/rfc8693/access_token_type_handler.go#L186